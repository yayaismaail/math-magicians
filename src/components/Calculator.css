.loading {
  margin-left: 20px;
  font-size: 1.5em;
  font-weight: 600;
}

.quote-title,
.quotes,
.author {
  display: flex;
  justify-content: center;
  align-items: center;
}

.quotes,
.author {
  font-size: 1.3em;
  width: 50%;
  margin: 0 auto;
  padding: 10px;
  text-align: center;
}

.Calc {
  margin: 12vh auto 0;
  display: flex;
  flex-direction: column;
  width: 35.5vw;
}

.Calc > * {
  display: flex;
}

.Display {
  display: flex;
  align-items: center;
  justify-content: flex-end;
  background-color: rgb(141, 141, 141);
  font-size: 2.2em;
  color: whitesmoke;
  padding-right: 20px;
  height: 60px;
  width: 460px;
}

.Calc-button {
  height: 60px;
  width: 120px;
  background-color: rgb(224, 224, 224);
  border: 1px solid rgb(184, 184, 184);
  font-size: 1.3em;
}

.Calc-button-zero {
  width: 240px;
  background-color: rgb(224, 224, 224);
  border: 1px solid rgb(184, 184, 184);
  font-size: 1.3em;
}

.Calc-operator {
  background-color: rgb(255, 145, 0);
}

.Button-row {
  display: flex;
  flex-direction: row;
}
