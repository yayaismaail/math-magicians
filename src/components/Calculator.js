import './Calculator.css';
import React, { useState } from 'react';
import calculate from '../logic/calculate';

function Calculator() {
  const [calculation, setCalculation] = useState({
    total: null,
    next: null,
    operation: null,
  });

  function handleClick(buttonName) {
    setCalculation(calculate(calculation, buttonName));
  }

  return (
    <>
      <h3 className="header">Let&apos;s do some math...</h3>
      <div className="Calc">
        <div className="row1 Display">{calculation.next || calculation.total || '0'}</div>
        <div className="row2">
          <button type="button" className="Calc-button" onClick={() => handleClick('AC')}>AC</button>
          <button type="button" className="Calc-button" onClick={() => handleClick('+/-')}>+/-</button>
          <button type="button" className="Calc-button" onClick={() => handleClick('%')}>%</button>
          <button type="button" className="Calc-button Calc-operator" onClick={() => handleClick('รท')}>รท</button>
        </div>
        <div className="row3">
          <button type="button" className="Calc-button" onClick={() => handleClick('7')}>7</button>
          <button type="button" className="Calc-button" onClick={() => handleClick('8')}>8</button>
          <button type="button" className="Calc-button" onClick={() => handleClick('9')}>9</button>
          <button type="button" className="Calc-button Calc-operator" onClick={() => handleClick('x')}>x</button>
        </div>
        <div className="row4">
          <button type="button" className="Calc-button" onClick={() => handleClick('4')}>4</button>
          <button type="button" className="Calc-button" onClick={() => handleClick('5')}>5</button>
          <button type="button" className="Calc-button" onClick={() => handleClick('6')}>6</button>
          <button type="button" className="Calc-button Calc-operator" onClick={() => handleClick('-')}>-</button>
        </div>
        <div className="row5">
          <button type="button" className="Calc-button" onClick={() => handleClick('1')}>1</button>
          <button type="button" className="Calc-button" onClick={() => handleClick('2')}>2</button>
          <button type="button" className="Calc-button" onClick={() => handleClick('3')}>3</button>
          <button type="button" className="Calc-button Calc-operator" onClick={() => handleClick('+')}>+</button>
        </div>
        <div className="row6">
          <button type="button" className="Calc-button-zero" onClick={() => handleClick('0')}>0</button>
          <button type="button" className="Calc-button" onClick={() => handleClick('.')}>.</button>
          <button type="button" className="Calc-button Calc-operator" onClick={() => handleClick('=')}>=</button>
        </div>
      </div>
    </>
  );
}

export default Calculator;
