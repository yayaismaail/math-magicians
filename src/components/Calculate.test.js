import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from './Calculator';

describe('Calculator', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('updates the display when buttons are clicked', () => {
    const { getByText, getByTestId } = render(<Calculator />);
    const display = getByTestId('display');

    fireEvent.click(getByText('1'));
    expect(display.textContent).toBe('1');

    fireEvent.click(getByText('+'));
    expect(display.textContent).toBe('1');

    fireEvent.click(getByText('2'));
    expect(display.textContent).toBe('2');

    fireEvent.click(getByText('='));
    expect(display.textContent).toBe('3');
  });
});
