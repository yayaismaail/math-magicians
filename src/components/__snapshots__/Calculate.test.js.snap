// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator matches the snapshot 1`] = `
<DocumentFragment>
  <h3
    class="header"
  >
    Let's do some math...
  </h3>
  <div
    class="Calc"
  >
    <div
      class="row1 Display"
    >
      0
    </div>
    <div
      class="row2"
    >
      <button
        class="Calc-button"
        type="button"
      >
        AC
      </button>
      <button
        class="Calc-button"
        type="button"
      >
        +/-
      </button>
      <button
        class="Calc-button"
        type="button"
      >
        %
      </button>
      <button
        class="Calc-button Calc-operator"
        type="button"
      >
        รท
      </button>
    </div>
    <div
      class="row3"
    >
      <button
        class="Calc-button"
        type="button"
      >
        7
      </button>
      <button
        class="Calc-button"
        type="button"
      >
        8
      </button>
      <button
        class="Calc-button"
        type="button"
      >
        9
      </button>
      <button
        class="Calc-button Calc-operator"
        type="button"
      >
        x
      </button>
    </div>
    <div
      class="row4"
    >
      <button
        class="Calc-button"
        type="button"
      >
        4
      </button>
      <button
        class="Calc-button"
        type="button"
      >
        5
      </button>
      <button
        class="Calc-button"
        type="button"
      >
        6
      </button>
      <button
        class="Calc-button Calc-operator"
        type="button"
      >
        -
      </button>
    </div>
    <div
      class="row5"
    >
      <button
        class="Calc-button"
        type="button"
      >
        1
      </button>
      <button
        class="Calc-button"
        type="button"
      >
        2
      </button>
      <button
        class="Calc-button"
        type="button"
      >
        3
      </button>
      <button
        class="Calc-button Calc-operator"
        type="button"
      >
        +
      </button>
    </div>
    <div
      class="row6"
    >
      <button
        class="Calc-button-zero"
        type="button"
      >
        0
      </button>
      <button
        class="Calc-button"
        type="button"
      >
        .
      </button>
      <button
        class="Calc-button Calc-operator"
        type="button"
      >
        =
      </button>
    </div>
  </div>
</DocumentFragment>
`;
